### PR TITLE
Add SMDDP RoBERTa notebook

### PR DESCRIPTION
**Pull request by Zha0q1**
## About
Adds fairseq RoBERTa example notebook.

Previously, https://github.com/aws/amazon-sagemaker-examples/pull/2964 did not get merged because the pinned commit leaks memory. Now that we have updated to the newest master in the original https://github.com/facebookresearch/fairseq repo, we are reopening this pr.

This notebook uses pt backend and the code change in fairseq is minimal (1 line only). ref: https://github.com/HerringForks/fairseq/tree/master_05_18_2022

## Some Benchmark Numbers  on P4d:
RoBERTa large (1.3B parameters), unit: words per second
**8 nodes:**  nccl backend 367385, smddp backend 417017 (**+13.5%**)
**16 nodes:** nccl backend 661451, smddp backend 827903 (**+25.2%**)

### CODE REVIEW

**Review by roywei — DISMISSED** [review]
LGTM!!

File: training/distributed_training/pytorch/data_parallel/roberta/roberta.ipynb [hunk 1]
```diff
@@ -0,0 +1,418 @@
+{
```
Comment (mchoi8739) [inline training/distributed_training/pytorch/data_parallel/roberta/roberta.ipynb:1]: <p>We don't support MXNet. Please remove MXNet from the intro sentence.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3420/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (mchoi8739) [inline training/distributed_training/pytorch/data_parallel/roberta/roberta.ipynb:1]: <p>Do you use PT 1.9 or PT 1.11? Please match the version number in #1.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3420/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (mchoi8739) [inline training/distributed_training/pytorch/data_parallel/roberta/roberta.ipynb:1]: <p>What's the model you use between the two (1.3B and 350M) by default in this notebook? Can we make if statement to pick up the right hyperparameters depending on an input string, something like </p><pre class="ql-syntax" spellcheck="false">if model_name=='roberta-l-1_3billion':
  hyperparameters= # the first hyperparameter dict
else:
  hyperparameters= # the second hyperparameter dict
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3420/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Comment by zhaoqizqwang** [comment]
Closing this PR since it was created a long time ago and didn't pass all the checks. Feel free to open a new pull request instead.

**Pull request closed without merging**